docs/source/reference/types.rst

```diff
@@ -102,6 +102,17 @@ the beginning or the end of the index specification::
    >>> numba.float32[::1, :, :]
    array(float32, 3d, F)
 
+This style of type declaration is supported within Numba :term:`JIT` compiled
+functions e.g. declaring the type of a :ref:`typed.List <feature-typed-list>`.
```

Comment:
I think this could do with a comma to separate the clauses of the sentence

```suggestion
functions, e.g. declaring the type of a :ref:`typed.List <feature-typed-list>`.
```